### ISSUE TITLE
Goes back to the default GDM theme.

### ISSUE THREAD

**Issue by valiantgenomics** [gdm-settings/gdm-settings#129]
First I want to say that this app is outstanding! And your work is highly highly appreciated!

I just wanted to note an issue I've been having on my:

- **OS: Pop!_OS 22.04 LTS x86_64** 
- **Kernel:  5.19.0-76051900-generic**

When I lock my screen or log out, it still shows me the default Gnome Display Manager (GDM) theme & settings.

And then in the bottom right corner I have a user icon. If I click on that user icon, only then the lock screen changes to my customised version, with my bespoke theme and background that I set via your Login Screen Manager app.

But the problem is that it always goes back to the default Gnome Display Manager (GDM) settings upon locking or logging out. And I have to click on the user icon to change it.

Maybe I can fix this by editing the gnome shell gsource file but I'm a bit hesitant to do so. I don't want to mess something up.

That's why I wanted to make you aware of this issue. Please, let me know if I can somehow help.

**Comment by realmazharhussain** [comment]
By 'user icon', do you mean the cogwheel icon? Could you send me a screenshot?

**Comment by valiantgenomics** [comment]
Thanks so much for the reply!

Yes, I tried getting a screenshot but I'm not sure how. Since I have to log out or lock my computer in order to get to the login screen. And at that point I can't access my screenshot app.

But I took a photo using my phone. Sorry it came out slightly blurry and I also blurred my username. But you can see in the bottom right corner the user icon I'm talking about.

It's not a cogwheel but it's a generic user icon. I even managed to find it (or near-identical one) online which I'm attaching below as well.

This is how my lock/login screen looks (I blurred my profile photo/name):

![image](https://user-images.githubusercontent.com/111633318/198036549-1cda18f5-861c-48ab-bd1c-e0cb84ecaa98.png)

And this is the near-identical icon I found online:

![userss](https://user-images.githubusercontent.com/111633318/198036950-b6030838-32de-4f05-bf8d-ae206b59dede.png)

**Comment by valiantgenomics** [comment]
Hey, yes when I click that user button in the bottom right corner, the entire screen goes black for 1-2 seconds and then it changes to the exact customized version that I set using your app. And then I can log in normally via the customized version.

But if I decide to lock or logout again later, it still goes to the default login screen first, not the customized one. So I have to repeat the same process by clicking the user icon in the bottom right corner.

**Comment by realmazharhussain** [comment]
Does the user button do anything other than that (e.g. Showing a pop-up where you choose a session)?

**Comment by valiantgenomics** [comment]
No, it doesn't show anything else. Even if I hover over with the mouse, it still doesn't show anything.

It only allows me to either left or right click it, and after that as I said above it opens the customized login screen.

**Comment by realmazharhussain** [comment]
That is weird.

**Comment by realmazharhussain** [comment]
I do not know where to go from here.

**Comment by valiantgenomics** [comment]
Yes, I probably have to edit the gnome-shell-theme.gresource file in order to make this work but I'm hesitant because doing so may break my system.

You can see when I go into my gnome-shell/theme root directory, I can only see the classic Pop theme. Which could be the reason why the customized theme doesn't load immediately and why the classic Pop theme loads by default.

I tried copying the theme I used for the customization into this root folder, but that didn't work.

![image](https://user-images.githubusercontent.com/111633318/198075595-fec88153-302b-43c9-9215-c302a3264b92.png)

**Comment by realmazharhussain** [comment]
You do not need to modify anything in `<datadir>/gnome-shell/theme` directory.

You also don't need to modify the `<datadir>/gnome-shell/gnome-shell-theme.gresource` file. On Ubuntu and similar systems (e.g. Pop!_OS), what you need to modify is `<datadir>/gnome-shell/gdm-theme.gresource` symlink target and the app already does that for you.

So, it seems like there is something wrong on Pop!_OS end i.e. with their patches to 'gdm3' package.

Or maybe Pop!_OS works differently than Ubuntu in this matter.

**Comment by valiantgenomics** [comment]
Thanks a lot for your help! I actually spoke with a very generous System76/PopOS developer who looked into this issue.

He essentially helped me switch to default GDM3 using the following commands:

```
mkdir gdm3
cd gdm3
git clone https://github.com/pop-os/gdm3
cd gdm3
sudo apt build-dep gdm3
git rebase -i HEAD~2
```

then change the word "pick" to "drop" infront of Pop patches
then save and close the file

```
dpkg-buildpackage -us -uc -b
cd ..
sudo dpkg -i *.deb
```

So I succesfully switched to default GDM3, updated and rebooted my system. However, this bug still persists.
Inferring from this, GDM3 might not be the issue.

You can also check PopOS's GDM3 here: https://github.com/pop-os/gdm3/commit/ea2c84f35f5877577610521e45f099d3380e0028

**Comment by realmazharhussain** [comment]
If the issue is really with GDM specifically on Pop!_OS, it probably is not with their GDM theme but something else.

Try installing installing [GDM3 package from Ubuntu 22.04](http://archive.ubuntu.com/ubuntu/pool/main/g/gdm3/gdm3_42.0-1ubuntu6_amd64.deb) and see if it works.

**Comment by valiantgenomics** [comment]
Thanks for the reply! Yes, this is what I meant. I installed the Ubuntu 22.04 GDM3 package. The PopOS dev told me how to switch from PopOS GDM3 to Ubuntu 22.04 GDM3. 

And I switched successfully but the issue still persists.

**Comment by realmazharhussain** [comment]
Try using [gdm-tools](https://github.com/realmazharhussain/gdm-tools) to change GDM theme and tell me if it works.

**Comment by valiantgenomics** [comment]
Hey, are you saying that **gdm-tools** is better than the **Login Manager Settings** (which is the GUI for gdm-tools)?

Because I'm already using the Login Manager Settings while having switched to Ubuntu 22.04 GDM3, but the issue still persists.

**Comment by realmazharhussain** [comment]
> Hey, are you saying that gdm-tools is better than the Login Manager Settings?

No.

> which is the GUI for gdm-tools

It is not a GUI for gdm-tools. It is just a GUI program that does what gdm-tools does plus more. It is an *alternative* to gdm-tools.

I want you to try using gdm-tools because it is a different program and if two different programs have this issue, it means the issue probably does not have anything to do with the app but your system configuration.

**Comment by valiantgenomics** [comment]
I see thanks for explaining. I downloaded gdm-tools, checked to see whether I have the dependencies and I did have them.

Followed the instructions and I believe I set the them correctly:

![image](https://user-images.githubusercontent.com/111633318/198998805-3ff74337-2331-4651-a0bc-4b06309f3385.png)

But the issue still persists. I am still on Ubuntu 22.04 GDM3.

Another thing I forgot to mention initially is that if I restart, or shut down and then power on my computer the Login screen that comes up is the correct, customized one via the Login Manager Settings app.

This issue seems to appear only on Lock/Logout but not on Restart/Turn off.

**Comment by Harmonia2Side** [comment]
I have run into this same issue using pop OS. However, using gdm-tools as suggested worked. I installed the latest version with the install script provided and ran
`set-gdm-theme -s default /path/to/image`
which then changed the background of the login screen as intended

**Comment by realmazharhussain** [comment]
@Harmonia2Side Have you tried using this app after that? Does the issue persist?

**Comment by Harmonia2Side** [comment]
I've just tried again, unfortunately it still does not appear to work. Uninstalling the app and reinstalling and even rebooting the whole system also had no effect.
Gdm-tools is already pretty satisfying for my use case, but I'm really curious about this issue, wondering what Pop OS does differently that the app doesn't work on it.

**Comment by realmazharhussain** [comment]
I haven't looked at gdm-tools's actual code in a long. I might need to revisit it to list out and test differences in logic of both of these pieces of software.

**Comment by Harmonia2Side** [comment]
I'll keep an eye on this thread, if there's something we can do to help look into this, do let us know :)

**Comment by realmazharhussain** [comment]
Does running the following commands fix the issue?

```bash
sudo cp /usr/share/gnome-shell/gnome-shell-theme.gresource.gdm_settings /usr/share/gnome-shell/gnome-shell-theme.gresource
sudo update-alternatives --set gdm-theme.gresource /usr/share/gnome-shell/gnome-shell-theme.gresource
```

**Comment by Harmonia2Side** [comment]
Thanks for looking into this, and apologies for the late reply.
I used the commands you suggested, and it works in an interesting kind of way.
If I do the suggested commands and change the wallpaper afterwards, as in:
->
```
sudo cp /usr/share/gnome-shell/gnome-shell-theme.gresource.gdm_settings /usr/share/gnome-shell/gnome-shell-theme.gresource
sudo update-alternatives --set gdm-theme.gresource /usr/share/gnome-shell/gnome-shell-theme.gresource
```
then 
-> Change wallpaper with gdm-setting (flatpak)

Nothing happens. But if I reverse the order, as in:
-> Change wallpaper with gdm-setting (flatpak), then
->
```
sudo cp /usr/share/gnome-shell/gnome-shell-theme.gresource.gdm_settings /usr/share/gnome-shell/gnome-shell-theme.gresource
sudo update-alternatives --set gdm-theme.gresource /usr/share/gnome-shell/gnome-shell-theme.gresource
```
Then the wallpaper is changed successfully. However, I have to redo the commands every time I change the wallpaper in gdm-settings, otherwise the wallpaper is not changed.

Also, this only works for "Pop" and "Pop-dark" themes. When using "default" and "default-pure" themes, it does not work at all.

**Comment by realmazharhussain** [comment]
Sorry for the long delay!

Can you post here the output of `grep ID /etc/os-release`, so that I can add a special case for Pop!_OS?

**Comment by Harmonia2Side** [comment]
> Sorry for the long delay!
> 
> Can you post here the output of `grep ID /etc/os-release`, so that I can add a special case for Pop!_OS?

Of course! Also apologies for the delay. Here it is:
```
$ grep ID /etc/os-release
```
```
ID=pop
ID_LIKE="ubuntu debian"
VERSION_ID="22.04"
```
Thank you very much for extending support to us as well :)

**Comment by realmazharhussain** [comment]
Someone has reported [a similar error](https://github.com/gdm-settings/gdm-settings/issues/158) on Fedora as well. Unless that bug comes out to be something unrelated, a special case for Pop!_OS might not be a good option anymore.

**Comment by Harmonia2Side** [comment]
Indeed, it might not. [It has been suggested](https://github.com/gdm-settings/gdm-settings/issues/158#issuecomment-1507644582) that the issue could be related to Gnome 44. POP!_OS is currently using Gnome 42.5. If the fedora issue ends up being due to Gnome 44, then a special case for POP might still be valid (speculating). But we indeed should wait and look further into the issue, as it might be possible to solve both simultaneously.

**Comment by Harmonia2Side** [comment]
> Someone has reported [a similar error](https://github.com/gdm-settings/gdm-settings/issues/158) on Fedora as well. Unless that bug comes out to be something unrelated, a special case for Pop!_OS might not be a good option anymore.

With the fedora issue being [solved sucessfuly](https://github.com/gdm-settings/gdm-settings/issues/158), do you know if there any relation between that issue and this one?
I have just tested the application again, and the option to change background image still does not seem to work.
Curiously, the option to add a logo below the user list is working.

**Comment by realmazharhussain** [comment]
Sorry, I had forgotten about this. It seems like this is a Pop!_OS specific issue.

**Comment by Harmonia2Side** [comment]
Apologies again for the long delay, this exam season was specially rough.
Do you think the fix you suggested [here](https://github.com/gdm-settings/gdm-settings/issues/129#issuecomment-1477628824) could work?

**Comment by Rid1-fz-06** [comment]
@realmazharhussain I don't know if this is the same issue, but every time I update mutter, gnome-shell or anything similar, the theme automatically resets. I open the login manager, and it says everything is OK. As a result, I always have to reset my settings. I'm currently using fedora 38. Is there any fix?

**Comment by realmazharhussain** [comment]
I am so sorry everyone that it took so long for this simple fix. But finally, here it is. I have added a special case for Pop!_OS. I just need to know that it works before merging and releasing it.

[gdm-settings.flatpak]
[Login_Manager_Settings.AppImage]

If anyone is building from source, the change I made lives in `3_pop_os_fix` branch.

[gdm-settings.flatpak]: https://mega.nz/file/HktGGAzS#cy6pnxV9NfSbt7FMFS81zhqF9xD8fFW5x_nIMLCRLjs
[Login_Manager_Settings.AppImage]: https://mega.nz/file/X0k2SJLZ#qni7wE0GDfRb1Ar3ohF0tplNTBdDOHAO1gzrmWE0LUI

<hr>

@Rid1-fz-06 That is actually a different issue. If you want, you can open a new issue tracker for it.

What happens there is that `/usr/share/gnome-shell/gnome-shell-theme.gresource` gets replaced by the default one on each update. So, you have to re-apply the settings. [gdm-tools], if you installed it on an Arch-based distro, had a solution for it. It was integrated with pacman so that on each update that required it, it would automatically re-apply the settings. But gdm-tools is deprecated now and I plan to add that feature to gdm-settings one day (but properly this time).

[gdm-tools]: https://github.com/realmazharhussain/gdm-tools

**Comment by Harmonia2Side** [comment]
> I am so sorry everyone that it took so long for this simple fix. But finally, here it is. I have added a special case for Pop!_OS. I just need to know that it works before merging and releasing it.\n\ngdm-settings.flatpak\nLogin_Manager_Settings.AppImage\n\nIf anyone is building from source, the change I made lives in 3_pop_os_fix branch.

Thank you very much for this fix, it was worth the wait. It's really exciting to have a fully functional login screen customizer.
I'll test the flatpak as soon as possible (probably over the next weekend or so) to make sure it works properly. Will answer back as soon as it does.

**Comment by VirtualJunky** [comment]
Wanted to add to this cause I'm having more issues then just the wallpaper not saving and they seem related but if not I can make a new issue. I am using Pop!_OS everything up to date, very few extensions installed. Originally tried the flatpak found in Pop!_Shop, but then found this topic and tried the special test version.

The other issue is when I used Display > "Apply current display settings" it made the resolution of the login screen terrible, like maybe even 480p if I had to guess. Every time I boot my PC now and login, it goes through 5+ seconds of the screen flickering black, presumably adjusting the resolution up to my normal 1440p, which is rather annoying.

Both issues happen in both of the flatpak versions I tried. Not really sure how to revert this change either. Also don't have the user button in the bottom right like OP does to see if it's applying elsewhere, never have, so unsure on what that is if that means anything.

I'm pretty new to Linux so it's very possible I did something wrong, just not sure what I could have done.

**Comment by realmazharhussain** [comment]
> Not really sure how to revert this change either.

```bash
sudo rm ~Debian-gdm/.config/monitors.xml
```

**Comment by realmazharhussain** [comment]
@VirtualJunky Try [this flatpak](https://github.com/gdm-settings/gdm-settings/files/12728701/gdm-settings.flatpak.zip).

**Comment by VirtualJunky** [comment]
> @VirtualJunky Try [this flatpak](https://github.com/gdm-settings/gdm-settings/files/12728701/gdm-settings.flatpak.zip).

Getting this error:
`error: The application io.github.realmazharhussain.GdmSettings/x86_64/master requires the runtime org.gnome.Platform/x86_64/45 which was not found`

Tried installing the normal flatpak version and then that one but same result, what am I missing?

**Comment by realmazharhussain** [comment]
> what am I missing?

> `org.gnome.Platform/x86_64/45 which was not found`

To install the missing platform version, run

```bash
flatpak install org.gnome.Platform//45
```

**Comment by Harmonia2Side** [comment]
> I just need to know that it works before merging and releasing it.
@realmazharhussain Hello there! Apologies for the long hiatus during the semester.
I am very glad to inform that the updated flatpak you provided works like a charm, the login screen is sucessfully updates on Pop!_OS 22.04 LTS. 

It took some time to figure out how to sucessfuly install the .flatpak file, I'll leave some tips on how to make that work soon for anyone who couldn't make it work.

Once again, thank you very much for bringing this to pop OS

**Comment by Harmonia2Side** [comment]
Explaining how I made the flatpak work here

> > @VirtualJunky Try [this flatpak](https://github.com/gdm-settings/gdm-settings/files/12728701/gdm-settings.flatpak.zip).
> 
> Getting this error: `error: The application io.github.realmazharhussain.GdmSettings/x86_64/master requires the runtime org.gnome.Platform/x86_64/45 which was not found`
> 
> Tried installing the normal flatpak version and then that one but same result, what am I missing?

@VirtualJunky I've run into the exact same issue as you. Tried to install the flatpak, received the error:

```
$ flatpak install ./gdm-settings.flatpak
error: The application io.github.realmazharhussain.GdmSettings/x86_64/master requires the runtime org.gnome.Platform/x86_64/44 which was not found

```
but if I tried installing it, the runtime was actually already installed
```
$ flatpak install flathub org.gnome.Platform/x86_64/44
Procurando por correspondências…
Ignorando: org.gnome.Platform/x86_64/44 já está instalado
```
Which made no sense at this point. Curiously, the system would ask for my password before trying to install, indicationg that it might be trying to install as root for all users. I then tried to install only for my user with:
```
flatpak install --user ./gdm-settings.flatpak
```
and then the installation was sucessful.
The app did not appear on the app menu, so I ran it with `flatpak run io.github.realmazharhussain.GdmSettings`. Everything worked after that.

My guess is that, for some reason, the root user is not recognizing the runtimes when trying to install. Anyway, now it works :)

**Comment by VirtualJunky** [comment]
> @VirtualJunky Try [this flatpak](https://github.com/gdm-settings/gdm-settings/files/12728701/gdm-settings.flatpak.zip).

Sorry been crazy few months, that version didn't work, however the first version you linked here https://github.com/gdm-settings/gdm-settings/issues/129#issuecomment-1694562534 does to an extent. Everything works now except the resolution issue I posted in my original comment here https://github.com/gdm-settings/gdm-settings/issues/129#issuecomment-1712990729 it says it has applied the settings but when I logout it's still real bad, again like 720p or maybe even worse.


> Explaining how I made the flatpak work here

Thank you for this comment, had to install it via --user as well and is how I got to try it again!

**Comment by realmazharhussain** [comment]
Hi, everyone. I have just realized that we were trying to fix the Lock screen, while this app is only made to customize the Login screen.

Fixing the lock screen is out of scope of this app, and user's regular preferences are automatically applied to the lock screen anyway. So, there's no need to even apply settings from this app.
